packages/story-editor/src/app/rightClickMenu/hooks/useIsPartiallyOffCanvas.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  DANGER_ZONE_HEIGHT,
+  PAGE_HEIGHT,
+  PAGE_WIDTH,
+} from '@googleforcreators/units';
+import { useCallback } from 'react';
+
+function useIsPartiallyOffCanvas(selectedElement) {
+  const { x, y, width, height } = selectedElement;
+
+  const isOffCanvas = useCallback(() => {
```

Comment:
Why is this a callback and not just a const (e.g. with `useMemo`)?

That said, since this custom hook here does not really use any other hooks (besides wrapping in `useCallback`, we could also just keep things simple and turn this logic into 2 regular functions, not using React at all.

Then we could move them to `src/utils` or elsewhere (since it's not tightly coupled to right click menu anyway)

Reply:
Good call - ya that code turned into `just regular functions` :)

Moved the functions out to `src/utils` and added tests.